syntax = "proto3";

option cc_enable_arenas = true;
package foobar;

service FooBar {
	rpc GetFoo(FooRequest) returns (FooResponse) {}
	rpc GetBar(BarRequest) returns (BarResponse) {}

	rpc SubscribeFoo (FooStreamRequest) returns (stream FooStreamNotification) {}
	rpc SubscribeBar (BarStreamRequest) returns (stream BarStreamNotification) {}

	rpc StreamFoo(stream FooStreamNotification) returns (FooResponse) {}
	rpc StreamBar(stream BarStreamNotification) returns (BarResponse) {}

	rpc ExchangeFooBar(stream FooStreamNotification) returns (stream BarStreamNotification) {}
}

message Foo {string value = 1;}
message Bar {int64 value = 1;}

message FooRequest {string value = 1;}
message FooResponse {Foo foo = 1;}

message BarRequest {string value = 1;}
message BarResponse {Bar bar = 1;}

message FooStreamRequest {string value = 1;}
message FooStreamNotification {Foo foo = 1;}

message BarStreamRequest {string value = 1;}
message BarStreamNotification {Bar bar = 1;}